### ISSUE TITLE
[Core]When you shield blocked damage - you get incorrect damage taken calculating 

### ISSUE THREAD

**Issue by sposobniy** [SunstriderEmu/BugTracker#1843]
Describe the bug: 
When you shield blocked damage - you get incorrect damage taken  calculating 

To Reproduce: 
1. Equip  (Fine Light Crossbow) http://tbc.cavernoftime.com/item=15808  with 29 based damage 
2.  Look at hits and crits in combat log , when you have  blocked damage - you take more damage than should.

My tests results : 
https://imgur.com/UTeLGs6  
(paladin have) 343 resilience here 

Another test  without few items 
https://imgur.com/P9aXQDx

**Comment by hufsa** [comment]
https://github.com/SunstriderEmu/BugTracker/issues/1839

**Comment by Jerrythemagicdog** [comment]
On first glance looks like resilience is not applied if a player blocks.

**Comment by sposobniy** [comment]
1) About ranged attacks like said Jerry , unaffected by resilience  
https://imgur.com/WTrFSQh test against non resilience target 

2) Melee blocked damage issue , looks much more wierd 
When you have block - looks like its unaffected by armor . 
For test used  http://tbc.cavernoftime.com/item=20983
1-2 damage dagger (tailents was reset) .
https://imgur.com/9WyNyKH  

And like link guy from closed report , https://user-images.githubusercontent.com/60829714/74106533-13421000-4b68-11ea-8f9a-7aa6e6e4563b.jpg  

When he had shield block he got 457 damage , before nonblocked hit was 265 .

**Comment by Jerrythemagicdog** [comment]
That report is not related, the damage done by the creature is random your example points out an actual problem.

**Comment by Rettiy** [comment]
To be honest further tests focused on whether Armor + Resilience (and crits) mitigate properly when applied at the same time can be reasonable and might be a reason why rogue and warrior damage is considered so abnormal.

**Comment by Tyrannize999** [comment]
It's okay the damage is random but still always the damage is not correct if the player blocks the hit. So I'm glad other player met the same problem, I don't think it's an incorrect report. For short my closed report: Like sposobniy said I got hits around 265 never went over 350 except crits. But always when I blocked, or used Shield Block I got 450 dmg. That's why I said, if it is not a bug, then what's the point of blocking dmg if the mob hits higher after using shield block ( and i'm not talking about crit dmg)?
